jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannelState.java

```diff
@@ -292,19 +293,13 @@ public String getStatusString()
         }
     }
 
-    public boolean completeResponse()
+    public Throwable completeResponse()
```

Comment:
The boolean return was giving protection against calling this method twice.
So either, there is a possibility that we will call it twice, in which case we still need the protection....
OR we will never call it twice, so instead we should throw ISE if we are ever called with it is already in COMPLETED state.

Reply:
This method is only called once because it is protected by the `ServletChannel` state machine.
From `ServletChannel.handle()`, we end up in the `TERMINATED` state where we call `onCompleted()` that calls this method.

Reply:
So can we have an assert (or ISE) if it is already in COMPLETED state

Reply:
Not sure what you refer to, as there are many COMPLETED states?
If you refer to `_outputState` it can also be ABORTED.
I feel like we don't need this extra assert, as we just want to make the transition OPEN -> COMPLETED.

Reply:
This method used to return a boolean that indicated if it had just transitioned from OPEN->COMPLETED.  If it returned false, then the callback would not be completed.

Now it returns a throwable to say if there has been a failure, in which case the callback is failed.   But it has lost the protection for multiple calls, so the callback might be called twice.   

I understand that you know this cannot happen because of external factors.... that nothing in this method represents that invariant.   It would be clearer if it had something like:
```java
    assert _failure != null || _outputState == OutputState.OPEN
```

If this assert does not hold, then this PR has changed behaviour in a non obvious way.